In [99]:
import pandas as pd
import numpy as np

# Data cleaning - literature dataset
## Import dataset

In [2]:
# load given dataset
lit_df = pd.read_csv('Literature-data_TU-Darmstadt.txt', sep = '\t', header = 1, skiprows = [3])
lit_df

Quelle                                              Autor  \
0       WOS  Chaudhary, Ayesha Hoor; Bukhari, Faisal; Iqbal...   
1       WOS  Kim, Soo-Kyun; Lee, Chang-Hee; Kim, Sun-Jeong;...   
2       WOS                     Lee, Byong Kwon; Lee, Yang Sun   
3       WOS  Frederiksen, Joakim Grant; Sorensen, Stine May...   
4       WOS      Rafique, Muhammad Usman; Cheung, Sen-ching S.   
5       WOS                     Checa, David; Bustillo, Andres   
6       WOS  Ammann, Jeanine; Hartmann, Christina; Peterhan...   
7       WOS  Curry, Christopher; Li, Ruixuan; Peterson, Nic...   
8       WOS  Lee, Sang Hyeok; Kim, Martha; Kim, Hyosun; Par...   
9       WOS  Ping, Jiamin; Weng, Dongdong; Liu, Yue; Wang, ...   
10      WOS  Fransson, Goran; Holmberg, Jorgen; Westelius, ...   
11      WOS  Appel, Lora; Appel, Eva; Bogler, Orly; Wiseman...   
12      WOS                      Griffin, Tom; Muldoon, Meghan   
13      WOS  Na, Woongsoo; Dao, Nhu-Ngoc; Kim, Joongheon; R...   
14      WOS  Tumialis, Alexey; Smirnov, Alexey; Fadeev, Kir...   
15      WOS                  Gu, Jiaxi; Yu, Zhiwen; Shen, Kele   
16      WOS  Zaleski-King, Ashley; Pinto, Robin; Lee, Gener...   
17      WOS    Hagita, Katsumi; Kodama, Yuuki; Takada, Masashi   
18      WOS             Bae, Yoosung; Cha, Baekdong; Ryu, Jeha   
19      WOS     Alrehaili, Enas Abdulrahman; Al Osman, Hussein   
20      WOS                 Greene, David; Groenendyk, Michael   
21      WOS  Jost, Tyler A.; Drewelow, Grant; Koziol, Scott...   
22      WOS                    Risi, Dante; Palmisano, Stephen   
23      WOS  Wang, Qiyue; Cheng, Yongchao; Jiao, Wenhua; Jo...   
24      WOS  Lubetzky, Anat V.; Harel, Daphna; Kelly, Jenni...   
25      WOS           Marchetto, Jonathan; Wright, W. Geoffrey   
26      WOS  Shu, Yu; Huang, Yen-Zhang; Chang, Shu-Hsuan; C...   
27      WOS  Klippel, Alexander; Zhao, Jiayan; Jackson, Kat...   
28      WOS  Cao, Shi; Nandakumar, Krithika; Babu, Raiju; T...   
29      WOS                    Kalarat, Kosin; Koomhin, Phanit   
...     ...                                                ...   
1039    BSP  Mühlberger, Andreas; Wieser, Matthias J.; Kenn...   
1040    BSP  Wren, Christopher R.; Sparacino, Flavia; Azarb...   
1041    BSP  Kesztyues, T.I.; Mehlitz, M.; Schilken, E.; We...   
1042    BSP                    Carnegie, Kieran; Rhee, Taehyun   
1043    BSP  Marner, Michael R.; Smith, Ross T.; Walsh, Jam...   
1044    BSP  Livatino, Salvatore; Muscato, Giovanni; Privit...   
1045    BSP  Livatino, Salvatore; De Paolis, Lucio T.; D'Ag...   
1046    BSP  Ruddle, Roy; Volkova, Ekaterina; Mohler, Betty...   
1047    BSP  Lott, Alison; Bisson, Etienne; Lajoie, Yves; M...   
1048    BSP  Cho, Baek-Hwan; Ku, Jeonghun; Jang, Dong Pyo; ...   
1049    BSP  Tse, Mimi M.Y.; Ng, Jacobus K.F.; Chung, Joann...   
1050    BSP                    Mania, Katerina; Chalmers, Alan   
1051    BSP  Rizzo, A.A.; Buckwalter, J.G.; Bowerly, T.; Va...   
1052    BSP            Jankowski, Jaros?aw; Grabowski, Andrzej   
1053    BSP  Nyberg, Lars; Lundin-Olsson, Lillemor; Sondell...   
1054    BSP   Barniv, Yair; Aguilar, Mario; Hasanbelliu, Erion   
1055    BSP                          Avila, Lisa; Bailey, Mike   
1056    BSP  Crosbie, J. H.; Lennon, S.; McNeill, M. D. J.;...   
1057    BSP  Schneider, Susan M.; Ellis, Mathew; Coombs, Wi...   
1058    BSP  Komura, Taku; Lau, Rynson W.H.; Lin, Ming C.; ...   
1059    BSP  Dahlquist, Lynnda M.; Herbert, Linda J.; Weiss...   
1060    BSP  Roberts, Jonathan C.; Ritsos, Panagiotis D.; B...   
1061    BSP       Encarnacao, L. Miguel; Barton III, Robert J.   
1062    BSP                                               ?,??   
1063     SB  Willemsen, P.; Colton, M.B.; Creem-Regehr, S.H...   
1064     SB  Grechkin, TY; Nguyen, TD; Plumert, JM; Cremer,...   
1065     SB  Hu, B; Ma, L; Zhang, W; Salvendy, G; Chablat, ...   
1066     SB  Taylor W. Cleworth, Brian C. Horslen, Mark G. ...   
1067     SB  Ouellet,É; Boller, B; Corriveau-

## Keep needed attributes

In [24]:
lit_clean = lit_df[['Autor', 'Titel', 'Jahr', 'Journal', 'Typ', 'DOI', 'Ausschlusspunkt']]
lit_clean

Autor  \
0     Chaudhary, Ayesha Hoor; Bukhari, Faisal; Iqbal...   
1     Kim, Soo-Kyun; Lee, Chang-Hee; Kim, Sun-Jeong;...   
2                        Lee, Byong Kwon; Lee, Yang Sun   
3     Frederiksen, Joakim Grant; Sorensen, Stine May...   
4         Rafique, Muhammad Usman; Cheung, Sen-ching S.   
5                        Checa, David; Bustillo, Andres   
6     Ammann, Jeanine; Hartmann, Christina; Peterhan...   
7     Curry, Christopher; Li, Ruixuan; Peterson, Nic...   
8     Lee, Sang Hyeok; Kim, Martha; Kim, Hyosun; Par...   
9     Ping, Jiamin; Weng, Dongdong; Liu, Yue; Wang, ...   
10    Fransson, Goran; Holmberg, Jorgen; Westelius, ...   
11    Appel, Lora; Appel, Eva; Bogler, Orly; Wiseman...   
12                        Griffin, Tom; Muldoon, Meghan   
13    Na, Woongsoo; Dao, Nhu-Ngoc; Kim, Joongheon; R...   
14    Tumialis, Alexey; Smirnov, Alexey; Fadeev, Kir...   
15                    Gu, Jiaxi; Yu, Zhiwen; Shen, Kele   
16    Zaleski-King, Ashley; Pinto, Robin; Lee, Gener...   
17      Hagita, Katsumi; Kodama, Yuuki; Takada, Masashi   
18               Bae, Yoosung; Cha, Baekdong; Ryu, Jeha   
19       Alrehaili, Enas Abdulrahman; Al Osman, Hussein   
20                   Greene, David; Groenendyk, Michael   
21    Jost, Tyler A.; Drewelow, Grant; Koziol, Scott...   
22                      Risi, Dante; Palmisano, Stephen   
23    Wang, Qiyue; Cheng, Yongchao; Jiao, Wenhua; Jo...   
24    Lubetzky, Anat V.; Harel, Daphna; Kelly, Jenni...   
25             Marchetto, Jonathan; Wright, W. Geoffrey   
26    Shu, Yu; Huang, Yen-Zhang; Chang, Shu-Hsuan; C...   
27    Klippel, Alexander; Zhao, Jiayan; Jackson, Kat...   
28    Cao, Shi; Nandakumar, Krithika; Babu, Raiju; T...   
29                      Kalarat, Kosin; Koomhin, Phanit   
...                                                 ...   
1039  Mühlberger, Andreas; Wieser, Matthias J.; Kenn...   
1040  Wren, Christopher R.; Sparacino, Flavia; Azarb...   
1041  Kesztyues, T.I.; Mehlitz, M.; Schilken, E.; We...   
1042                    Carnegie, Kieran; Rhee, Taehyun   
1043  Marner, Michael R.; Smith, Ross T.; Walsh, Jam...   
1044  Livatino, Salvatore; Muscato, Giovanni; Privit...   
1045  Livatino, Salvatore; De Paolis, Lucio T.; D'Ag...   
1046  Ruddle, Roy; Volkova, Ekaterina; Mohler, Betty...   
1047  Lott, Alison; Bisson, Etienne; Lajoie, Yves; M...   
1048  Cho, Baek-Hwan; Ku, Jeonghun; Jang, Dong Pyo; ...   
1049  Tse, Mimi M.Y.; Ng, Jacobus K.F.; Chung, Joann...   
1050                    Mania, Katerina; Chalmers, Alan   
1051  Rizzo, A.A.; Buckwalter, J.G.; Bowerly, T.; Va...   
1052            Jankowski, Jaros?aw; Grabowski, Andrzej   
1053  Nyberg, Lars; Lundin-Olsson, Lillemor; Sondell...   
1054   Barniv, Yair; Aguilar, Mario; Hasanbelliu, Erion   
1055                          Avila, Lisa; Bailey, Mike   
1056  Crosbie, J. H.; Lennon, S.; McNeill, M. D. J.;...   
1057  Schneider, Susan M.; Ellis, Mathew; Coombs, Wi...   
1058  Komura, Taku; Lau, Rynson W.H.; Lin, Ming C.; ...   
1059  Dahlquist, Lynnda M.; Herbert, Linda J.; Weiss...   
1060  Roberts, Jonathan C.; Ritsos, Panagiotis D.; B...   
1061       Encarnacao, L. Miguel; Barton III, Robert J.   
1062                                               ?,??   
1063  Willemsen, P.; Colton, M.B.; Creem-Regehr, S.H...   
1064  Grechkin, TY; Nguyen, TD; Plumert, JM; Cremer,...   
1065  Hu, B; Ma, L; Zhang, W; Salvendy, G; Chablat, ...   
1066  Taylor W. Cleworth, Brian C. Horslen, Mark G. ...   
1067  Ouellet,É; Boller, B; Corriveau-Lecavalier, N;...   
1068                     Li, Z; Phillips, J; Durgin, FH   

                                                  Titel  Jahr  \
0        Laparoscopic Training Exercises Using HTC VIVE  2020   
1     Implementation of Local Area VR Environment us...  2020   
2     Distinction Between Real Faces and Photos by A...  2020   
3     Cognitive load and performance in immersive vi...  2020   
4           Tracking Attacks on Virtual Reality Systems  2020   
5

## Convert some string columns to lowercase

In [25]:
lowercase_attr = ['Autor', 'Titel', 'Journal']
for i in lowercase_attr:
    lit_clean.loc[:, i] = lit_clean.loc[:, i].str.lower()
    
lit_clean

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\indexing.py:543: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item] = s


Autor  \
0     chaudhary, ayesha hoor; bukhari, faisal; iqbal...   
1     kim, soo-kyun; lee, chang-hee; kim, sun-jeong;...   
2                        lee, byong kwon; lee, yang sun   
3     frederiksen, joakim grant; sorensen, stine may...   
4         rafique, muhammad usman; cheung, sen-ching s.   
5                        checa, david; bustillo, andres   
6     ammann, jeanine; hartmann, christina; peterhan...   
7     curry, christopher; li, ruixuan; peterson, nic...   
8     lee, sang hyeok; kim, martha; kim, hyosun; par...   
9     ping, jiamin; weng, dongdong; liu, yue; wang, ...   
10    fransson, goran; holmberg, jorgen; westelius, ...   
11    appel, lora; appel, eva; bogler, orly; wiseman...   
12                        griffin, tom; muldoon, meghan   
13    na, woongsoo; dao, nhu-ngoc; kim, joongheon; r...   
14    tumialis, alexey; smirnov, alexey; fadeev, kir...   
15                    gu, jiaxi; yu, zhiwen; shen, kele   
16    zaleski-king, ashley; pinto, robin; lee, gener...   
17      hagita, katsumi; kodama, yuuki; takada, masashi   
18               bae, yoosung; cha, baekdong; ryu, jeha   
19       alrehaili, enas abdulrahman; al osman, hussein   
20                   greene, david; groenendyk, michael   
21    jost, tyler a.; drewelow, grant; koziol, scott...   
22                      risi, dante; palmisano, stephen   
23    wang, qiyue; cheng, yongchao; jiao, wenhua; jo...   
24    lubetzky, anat v.; harel, daphna; kelly, jenni...   
25             marchetto, jonathan; wright, w. geoffrey   
26    shu, yu; huang, yen-zhang; chang, shu-hsuan; c...   
27    klippel, alexander; zhao, jiayan; jackson, kat...   
28    cao, shi; nandakumar, krithika; babu, raiju; t...   
29                      kalarat, kosin; koomhin, phanit   
...                                                 ...   
1039  mühlberger, andreas; wieser, matthias j.; kenn...   
1040  wren, christopher r.; sparacino, flavia; azarb...   
1041  kesztyues, t.i.; mehlitz, m.; schilken, e.; we...   
1042                    carnegie, kieran; rhee, taehyun   
1043  marner, michael r.; smith, ross t.; walsh, jam...   
1044  livatino, salvatore; muscato, giovanni; privit...   
1045  livatino, salvatore; de paolis, lucio t.; d'ag...   
1046  ruddle, roy; volkova, ekaterina; mohler, betty...   
1047  lott, alison; bisson, etienne; lajoie, yves; m...   
1048  cho, baek-hwan; ku, jeonghun; jang, dong pyo; ...   
1049  tse, mimi m.y.; ng, jacobus k.f.; chung, joann...   
1050                    mania, katerina; chalmers, alan   
1051  rizzo, a.a.; buckwalter, j.g.; bowerly, t.; va...   
1052            jankowski, jaros?aw; grabowski, andrzej   
1053  nyberg, lars; lundin-olsson, lillemor; sondell...   
1054   barniv, yair; aguilar, mario; hasanbelliu, erion   
1055                          avila, lisa; bailey, mike   
1056  crosbie, j. h.; lennon, s.; mcneill, m. d. j.;...   
1057  schneider, susan m.; ellis, mathew; coombs, wi...   
1058  komura, taku; lau, rynson w.h.; lin, ming c.; ...   
1059  dahlquist, lynnda m.; herbert, linda j.; weiss...   
1060  roberts, jonathan c.; ritsos, panagiotis d.; b...   
1061       encarnacao, l. miguel; barton iii, robert j.   
1062                                               ?,??   
1063  willemsen, p.; colton, m.b.; creem-regehr, s.h...   
1064  grechkin, ty; nguyen, td; plumert, jm; cremer,...   
1065  hu, b; ma, l; zhang, w; salvendy, g; chablat, ...   
1066  taylor w. cleworth, brian c. horslen, mark g. ...   
1067  ouellet,é; boller, b; corriveau-lecavalier, n;...   
1068                     li, z; phillips, j; durgin, fh   

                                                  Titel  Jahr  \
0        laparoscopic training exercises using htc vive  2020   
1     implementation of local area vr environment us...  2020   
2     distinction between real faces and photos by a...  2020   
3     cognitive load and performance in immersive vi...  2020   
4           tracking attacks on virtual reality systems  2020   
5

## Remove duplicates
### Rows with duplicate titles

In [26]:
lit_clean[lit_clean['Titel'].duplicated()]

Autor  \
800                             riva, g; wiederhold, bk   
1012  simpson, gordon; johnston, lucy; richardson, m...   
1059  dahlquist, lynnda m.; herbert, linda j.; weiss...   

                                                  Titel  Jahr  \
800   the new dawn of virtual reality in health care...  2015   
1012  an investigation of road crossing in a virtual...  2003   
1059  virtual-reality distraction and cold-pressor p...  2010   

                                                Journal  \
800   annual review of cybertherapy and telemedicine...   
1012                     accident analysis & prevention   
1059      cyberpsychology, behavior & social networking   

                        Typ                            DOI Ausschlusspunkt  
800   Article; Book Chapter    10.3233/978-1-61499-595-1-3        Dopplung  
1012                Article  10.1016/S0001-4575(02)00081-7        Dopplung  
1059                Article        10.1089/cyber.2009.0263        Dopplung

We find that row 782 and 800 are duplicates, but the DOI is missing in row 782, so we decide to keep row 800

In [28]:
# 782 and 800 are duplicates
lit_clean[lit_clean['Titel'] == lit_clean.iloc[800]['Titel']]

Autor  \
782  riva, g; wiederhold, bk   
800  riva, g; wiederhold, bk   

                                                 Titel  Jahr  \
782  the new dawn of virtual reality in health care...  2015   
800  the new dawn of virtual reality in health care...  2015   

                                               Journal                    Typ  \
782     annual review of cybertherapy and telemedicine                Article   
800  annual review of cybertherapy and telemedicine...  Article; Book Chapter   

                             DOI Ausschlusspunkt  
782                          NaN        Abstract  
800  10.3233/978-1-61499-595-1-3        Dopplung

In [31]:
lit_clean.at[800, 'Ausschlusspunkt'] = lit_clean.at[782, 'Ausschlusspunkt']

In [29]:
# 942 and 1012 are duplicates
lit_clean[lit_clean['Titel'] == lit_clean.iloc[1012]['Titel']]

Autor  \
942              simpson, g; johnston, l; richardson, m   
1012  simpson, gordon; johnston, lucy; richardson, m...   

                                                  Titel  Jahr  \
942   an investigation of road crossing in a virtual...  2003   
1012  an investigation of road crossing in a virtual...  2003   

                               Journal      Typ  \
942   accident analysis and prevention  Article   
1012    accident analysis & prevention  Article   

                                DOI Ausschlusspunkt  
942   10.1016/S0001-4575(02)00081-7        Abstract  
1012  10.1016/S0001-4575(02)00081-7        Dopplung

In [30]:
# 860 and 1059 are duplicates
lit_clean[lit_clean['Titel'] == lit_clean.iloc[1059]['Titel']]

Autor  \
860    dahlquist, lm; herbert, lj; weiss, ke; jimeno, m   
1059  dahlquist, lynnda m.; herbert, linda j.; weiss...   

                                                  Titel  Jahr  \
860   virtual-reality distraction and cold-pressor p...  2010   
1059  virtual-reality distraction and cold-pressor p...  2010   

                                             Journal      Typ  \
860   cyberpsychology behavior and social networking  Article   
1059   cyberpsychology, behavior & social networking  Article   

                          DOI Ausschlusspunkt  
860   10.1089/cyber.2009.0263        Abstract  
1059  10.1089/cyber.2009.0263        Dopplung

In [33]:
# remove rows 782, 1012, 1059
lit_clean = lit_clean.drop([782, 1012, 1059]).reset_index(drop = True)

### Rows with duplicate DOIs (not NaN or 0)

In [42]:
x = lit_clean[lit_clean['DOI'].duplicated()]
x = x[(x['DOI'] != '0') & (x['DOI'].notnull())]
x

Autor  \
667   liang, hai-ning; lu, feiyu; shi, yuwei; nanjap...   
668   somrak, andrej; humar, iztok; hossain, m. sham...   
669         newbutt, nigel; bradley, ryan; conley, iian   
670               aldaba, cassandra n.; moussavi, zahra   
671                feldstein, ilja t.; dyszak, georg n.   
673           park, kyudong; kim, dohyeon; han, sung h.   
674   martínez-navarro, jesus; bigné, enrique; guixe...   
675   sportillo, daniele; paljic, alexis; ojeda, luc...   
676                       czub, marcin; piskorz, joanna   
677   flavián, carlos; ibáñez-sánchez, sergio; orús,...   
678   louison, céphise; ferlay, fabien; mestre, dani...   
680   guna, jože; geršak, gregor; humar, iztok; song...   
682          erra, ugo; malandrino, delfina; pepe, luca   
683   blissing, björn; bruzelius, fredrik; eriksson,...   
685   laurell, christofer; sandström, christian; ber...   
686     farah, maya f.; ramadan, zahy b.; harb, dana h.   
687                     chittaro, luca; sioni, riccardo   
688   weier, martin; roth, thorsten; kruijff, ernst;...   
691   almousa, omamah; prates, joana; yeslam, noor; ...   
692   grudzewski, filip; awdziej, marcin; mazurek, g...   
693   frost, megan; goates, michael; cheng, sarah; j...   
694   tham, jason; duin, ann hill; gee, laura; ernst...   
695   katsigiannis, stamos; willis, rhys; ramzan, naeem   
696   baker, steven; waycott, jenny; robertson, elen...   
697   grubert, jens; ofek, eyal; pahud, michel; kris...   
698   zhang, yingxue; wang, yingbin; liu, feiyang; l...   
700               loup, guillaume; loup-escande, emilie   
701   chessa, manuela; maiello, guido; borsari, ales...   
702   han, ping-hsuan; chen, yang-sheng; liu, iou-sh...   
703   lombart, cindy; millan, elena; normand, jean-m...   
...                                                 ...   
729                                    cochrane, thomas   
731   stupar-rutenfrans, snežana; ketelaars, loes e....   
733        jeong, seongwook; jung, eui s.; im, youngjae   
735   starrett, michael j.; stokes, jared d.; huffma...   
736   wiederhold, mark d.; crisci, megan; patel, vra...   
737                    preuss, nora; ehrsson, h. henrik   
762   peukert, c; pfeiffer, j; meissner, m; pfeiffer...   
1007  ferrari, vincenzo; megali, giuseppe; troia, el...   
1009  baumann, stephen; neff, chris; fetzick, scott;...   
1011  kim, kwanguk; kim, chan-hyung; cha, kyung ryeo...   
1017  girard, benoit; turcotte, vincent; bouchard, s...   
1019  kang, youn joo; jeonghun ku; kiwan han; sun i....   
1023  gold, jeffrey i.; seok hyeon kim; kant, alexis...   
1024  lin, chiuhsiang joe; chen, hung?jen; cheng, pi...   
1025           marsh, tim; wright, peter; smith, shamus   
1027  hoffman, hunter g.; meyer, walter j.; ramirez,...   
1028  simone, lisa k.; schultheis, maria t.; rebimba...   
1031                yang, ungyeon; kim, gerardjounghyun   
1037  mühlberger, andreas; wieser, matthias j.; kenn...   
1039  kesztyues, t.i.; mehlitz, m.; schilken, e.; we...   
1042  livatino, salvatore; muscato, giovanni; privit...   
1043  livatino, salvatore; de paolis, lucio t.; d'ag...   
1044  ruddle, roy; volkova, ekaterina; mohler, betty...   
1045  lott, alison; bisson, etienne; lajoie, yves; m...   
1046  cho, baek-hwan; ku, jeonghun; jang, dong pyo; ...   
1049  rizzo, a.a.; buckwalter, j.g.; bowerly, t.; va...   
1050            jankowski, jaros?aw; grabowski, andrzej   
1051  nyberg, lars; lundin-olsson, lillemor; sondell...   
1054  crosbie, j. h.; lennon, s.; mcneill, m. d. j.;...   
1055  schneider, susan m.; ellis, mathew; coombs, wi...   

                                                  Titel  Jahr  \
667   evaluating the effects of collaboration and co...  2019   
668   estimating vr sickness and user experience usi...  2019   
669   using virtual reality head-mounted displays in...  2020   
670   effects of virtual reality technology locomoti...  2020   
671   road crossing decisions in real and virtual en...  2020   
6

In [44]:
# remove rows with duplicate DOIs
to_drop = list(x.index.values) # a list like so: [667, 668, ..., 1054, 1055]
lit_clean = lit_clean.drop(to_drop).reset_index(drop = True)

### Remaining rows with Ausschlusspunkt = Dopplung

In [46]:
lit_clean[lit_clean['Ausschlusspunkt'] == 'Dopplung']

Autor  \
666  peukert, christian; pfeiffer, jella; meißner, ...   
681                      gleasure, rob; feller, joseph   
970  hadipriono, fabian c.; duane, josann w.; nemet...   
973  baek-hwan cho; kim, saebyul; dong ik shin; jan...   

                                                 Titel  Jahr  \
666  shopping in virtual reality stores: the influe...  2019   
681  a rift in the ground: theorizing the evolution...  2016   
970  implementation of a virtual environment for tr...  2003   
973  neurofeedback training with virtual reality fo...  2004   

                                               Journal      Typ  \
666          journal of management information systems      NaN   
681  journal of the association for information sys...      NaN   
970             journal of intelligent & fuzzy systems  Article   
973                         cyberpsychology & behavior  Article   

                               DOI Ausschlusspunkt  
666  10.1080/07421222.2019.1628889        Dopplung  
681                              0        Dopplung  
970                            NaN        Dopplung  
973       10.1089/1094931042403055        Dopplung

For the remaining 4 rows marked as 'Dopplung', duplicates couldn't be found for rows 666 and 973, so we keep these two rows.

In [61]:
lit_clean = lit_clean.drop([681, 970]).reset_index(drop = True)

## Other modifications
We found rows with DOIs of 0 and NaN. Since our project idea is about how useful an article is based on the impact factor of its publishing journal, an invalid DOI is less relevant, so we decided to convert all null values to '0' for consistency instead of deleting those rows.

In [69]:
print(str(len(lit_clean[lit_clean['DOI'] == '0'])) + ' rows with DOI = 0')
print(str(len(lit_clean[lit_clean['DOI'].isnull()])) + ' rows with DOI = NaN')

28 rows with DOI = 0
29 rows with DOI = NaN


In [70]:
lit_clean.loc[lit_clean['DOI'].isnull(), 'DOI'] = '0'

One of the values in the Typ column is 'Artikle'. We changed it to 'Article'.

In [75]:
lit_clean[lit_clean['Typ'] == 'Artikle']

Autor  \
986  grechkin, ty; nguyen, td; plumert, jm; cremer,...   

                                                 Titel  Jahr  \
986  how does presentation method and measurement p...  2010   

                                    Journal      Typ                      DOI  \
986  acm transactions on applied perception  Artikle  10.1145/1823738.1823744   

    Ausschlusspunkt  
986          Sample

In [76]:
lit_clean.loc[lit_clean['Typ'] == 'Artikle', 'Typ'] = 'Article'

## Export as csv file

In [77]:
lit_clean

Autor  \
0    chaudhary, ayesha hoor; bukhari, faisal; iqbal...   
1    kim, soo-kyun; lee, chang-hee; kim, sun-jeong;...   
2                       lee, byong kwon; lee, yang sun   
3    frederiksen, joakim grant; sorensen, stine may...   
4        rafique, muhammad usman; cheung, sen-ching s.   
5                       checa, david; bustillo, andres   
6    ammann, jeanine; hartmann, christina; peterhan...   
7    curry, christopher; li, ruixuan; peterson, nic...   
8    lee, sang hyeok; kim, martha; kim, hyosun; par...   
9    ping, jiamin; weng, dongdong; liu, yue; wang, ...   
10   fransson, goran; holmberg, jorgen; westelius, ...   
11   appel, lora; appel, eva; bogler, orly; wiseman...   
12                       griffin, tom; muldoon, meghan   
13   na, woongsoo; dao, nhu-ngoc; kim, joongheon; r...   
14   tumialis, alexey; smirnov, alexey; fadeev, kir...   
15                   gu, jiaxi; yu, zhiwen; shen, kele   
16   zaleski-king, ashley; pinto, robin; lee, gener...   
17     hagita, katsumi; kodama, yuuki; takada, masashi   
18              bae, yoosung; cha, baekdong; ryu, jeha   
19      alrehaili, enas abdulrahman; al osman, hussein   
20                  greene, david; groenendyk, michael   
21   jost, tyler a.; drewelow, grant; koziol, scott...   
22                     risi, dante; palmisano, stephen   
23   wang, qiyue; cheng, yongchao; jiao, wenhua; jo...   
24   lubetzky, anat v.; harel, daphna; kelly, jenni...   
25            marchetto, jonathan; wright, w. geoffrey   
26   shu, yu; huang, yen-zhang; chang, shu-hsuan; c...   
27   klippel, alexander; zhao, jiayan; jackson, kat...   
28   cao, shi; nandakumar, krithika; babu, raiju; t...   
29                     kalarat, kosin; koomhin, phanit   
..                                                 ...   
961  steed, anthony; steptoe, william; oyekoya, wol...   
962                 gogouvitis, x. v.; vosniakos, g-c.   
963                                      navab, nassir   
964                   kesavadas, t.; subramanium, hari   
965       jun, hyun-kyu; dong-hyung lee; doh-chul yang   
966  geuss, michael n.; stefanucci, jeanine k.; cre...   
967      mikropoulos, tassos a.; strouboulis, vassilis   
968  baños, r. m.; botella, c.; alcañiz, m.; liaño,...   
969                                            noname,   
970  tamura, hideyuki; yamamoto, hiroyuki; katayama...   
971  baek-hwan cho; kim, saebyul; dong ik shin; jan...   
972                conradi, jessica; alexander, thomas   
973      ott, renaud; thalmann, daniel; vexo, frédéric   
974  wren, christopher r.; sparacino, flavia; azarb...   
975                    carnegie, kieran; rhee, taehyun   
976  marner, michael r.; smith, ross t.; walsh, jam...   
977  tse, mimi m.y.; ng, jacobus k.f.; chung, joann...   
978                    mania, katerina; chalmers, alan   
979   barniv, yair; aguilar, mario; hasanbelliu, erion   
980                          avila, lisa; bailey, mike   
981  komura, taku; lau, rynson w.h.; lin, ming c.; ...   
982  roberts, jonathan c.; ritsos, panagiotis d.; b...   
983       encarnacao, l. miguel; barton iii, robert j.   
984                                               ?,??   
985  willemsen, p.; colton, m.b.; creem-regehr, s.h...   
986  grechkin, ty; nguyen, td; plumert, jm; cremer,...   
987  hu, b; ma, l; zhang, w; salvendy, g; chablat, ...   
988  taylor w. cleworth, brian c. horslen, mark g. ...   
989  ouellet,é; boller, b; corriveau-lecavalier, n;...   
990                     li, z; phillips, j; durgin, fh   

                                                 Titel  Jahr  \
0       laparoscopic training exercises using htc vive  2020   
1    implementation of local area vr environment us...  2020   
2    distinction between real faces and photos by a...  2020   
3    cognitive load and performance in immersive vi...  2020   
4          tracking attacks on virtual reality systems  2020   
5    advantages and limits of virtual reality in le...  2020   
6   

In [80]:
wanted_attr = ['Titel', 'Autor', 'Jahr', 'Journal', 'Typ', 'DOI']
lit_clean[wanted_attr].to_csv('literature_dataset.csv', index = False)

# Data cleaning - impact factor dataset
## Import dataset

In [3]:
# load dataset of journal impact factors
# from https://www.kaggle.com/datasets/mostafafaramin/scientific-journals-ranking-sjr
journal_df = pd.read_csv('journal_impact_factors_2020.csv')
journal_df

Rank                                 Full Journal Title  \
0          1                 CA-A CANCER JOURNAL FOR CLINICIANS   
1          2                    NEW ENGLAND JOURNAL OF MEDICINE   
2          3                           Nature Reviews Materials   
3          4                      NATURE REVIEWS DRUG DISCOVERY   
4          5                                             LANCET   
5          6                        WHO Technical Report Series   
6          7              NATURE REVIEWS MOLECULAR CELL BIOLOGY   
7          8                   Nature Reviews Clinical Oncology   
8          9                              NATURE REVIEWS CANCER   
9         10                                   CHEMICAL REVIEWS   
10        11                                      Nature Energy   
11        12   JAMA-JOURNAL OF THE AMERICAN MEDICAL ASSOCIATION   
12        13                          REVIEWS OF MODERN PHYSICS   
13        14                           CHEMICAL SOCIETY REVIEWS   
14        15                                             NATURE   
15        16                                            SCIENCE   
16        17                     Nature Reviews Disease Primers   
17        18                                   World Psychiatry   
18        18                                   World Psychiatry   
19        20                          NATURE REVIEWS IMMUNOLOGY   
20        21                                   NATURE MATERIALS   
21        22                                               CELL   
22        23                               NATURE BIOTECHNOLOGY   
23        24                                    NATURE MEDICINE   
24        25                       Living Reviews in Relativity   
25        26                           Nature Reviews Chemistry   
26        27                        NATURE REVIEWS MICROBIOLOGY   
27        28                                    LANCET ONCOLOGY   
28        29                        NATURE REVIEWS NEUROSCIENCE   
29        30                            NATURE REVIEWS GENETICS   
...      ...                                                ...   
12825  12826  Correspondances en Metabolismes Hormones Diabe...   
12826  12827                                    NATURAL HISTORY   
12827  12828                               ACTA PHYSICA SLOVACA   
12828  12828                        BELL LABS TECHNICAL JOURNAL   
12829  12828     Bollettino di Storia delle Scienze Matematiche   
12830  12828              Current Perspectives in Social Theory   
12831  12828           Journal of Neurological Sciences-Turkish   
12832  12828                                PULP & PAPER-CANADA   
12833  12828                                    SEN-I GAKKAISHI   
12834  12828                     SURFACE COATINGS INTERNATIONAL   
12835  12828                                            Scandia   
12836  12828                   TIJDSCHRIFT VOOR DIERGENEESKUNDE   
12837  12838                       Advancements of Microbiology   
12838  12838                                      BMC Chemistry   
12839  12838                     BMC Molecular and Cell Biology   
12840  12838                                    Biologia Futura   
12841  12838                    Brazilian Journal of Psychiatry   
12842  12838                    Brazilian Journal of Psychiatry   
12843  12838                        Electronic Research Archive   
12844  12838                           Environmental Microbiome   
12845  12838       Environmental Pollutants and Bioavailability   
12846  12838                   Food Science of Animal Resources   
12847  12838                                       Geochemistry   
12848  12838                      Journal of Computer Languages   
12849  12838                    Mining Metallurgy & Exploration   
12850  12838  Photobiomodulation Photomedicine and Laser Sur...   
12851  12838          Polymer-Plastics Technology and Materials   
12852  12838             Sexual and Reproductive Health

## Keep needed attributes

In [105]:
rename_attr = {
    'Full Journal Title': 'Title',
    'Rank': 'Rank',
    'Journal Impact Factor': 'Impact_Factor',
    '5-Year Impact Factor': '5_Yr_IF',
    'Cited Half-Life': 'Half_Life',
    '% Articles in Citable Items': 'Percentage_Citable',
    'Average Journal Impact Factor Percentile': 'Avg_IF_Percentile'
}
journal_clean = journal_df.rename(columns = rename_attr)[list(rename_attr.values())]
journal_clean

Title   Rank  \
0                     CA-A CANCER JOURNAL FOR CLINICIANS      1   
1                        NEW ENGLAND JOURNAL OF MEDICINE      2   
2                               Nature Reviews Materials      3   
3                          NATURE REVIEWS DRUG DISCOVERY      4   
4                                                 LANCET      5   
5                            WHO Technical Report Series      6   
6                  NATURE REVIEWS MOLECULAR CELL BIOLOGY      7   
7                       Nature Reviews Clinical Oncology      8   
8                                  NATURE REVIEWS CANCER      9   
9                                       CHEMICAL REVIEWS     10   
10                                         Nature Energy     11   
11      JAMA-JOURNAL OF THE AMERICAN MEDICAL ASSOCIATION     12   
12                             REVIEWS OF MODERN PHYSICS     13   
13                              CHEMICAL SOCIETY REVIEWS     14   
14                                                NATURE     15   
15                                               SCIENCE     16   
16                        Nature Reviews Disease Primers     17   
17                                      World Psychiatry     18   
18                                      World Psychiatry     18   
19                             NATURE REVIEWS IMMUNOLOGY     20   
20                                      NATURE MATERIALS     21   
21                                                  CELL     22   
22                                  NATURE BIOTECHNOLOGY     23   
23                                       NATURE MEDICINE     24   
24                          Living Reviews in Relativity     25   
25                              Nature Reviews Chemistry     26   
26                           NATURE REVIEWS MICROBIOLOGY     27   
27                                       LANCET ONCOLOGY     28   
28                           NATURE REVIEWS NEUROSCIENCE     29   
29                               NATURE REVIEWS GENETICS     30   
...                                                  ...    ...   
12825  Correspondances en Metabolismes Hormones Diabe...  12826   
12826                                    NATURAL HISTORY  12827   
12827                               ACTA PHYSICA SLOVACA  12828   
12828                        BELL LABS TECHNICAL JOURNAL  12828   
12829     Bollettino di Storia delle Scienze Matematiche  12828   
12830              Current Perspectives in Social Theory  12828   
12831           Journal of Neurological Sciences-Turkish  12828   
12832                                PULP & PAPER-CANADA  12828   
12833                                    SEN-I GAKKAISHI  12828   
12834                     SURFACE COATINGS INTERNATIONAL  12828   
12835                                            Scandia  12828   
12836                   TIJDSCHRIFT VOOR DIERGENEESKUNDE  12828   
12837                       Advancements of Microbiology  12838   
12838                                      BMC Chemistry  12838   
12839                     BMC Molecular and Cell Biology  12838   
12840                                    Biologia Futura  12838   
12841                    Brazilian Journal of Psychiatry  12838   
12842                    Brazilian Journal of Psychiatry  12838   
12843                        Electronic Research Archive  12838   
12844                           Environmental Microbiome  12838   
12845       Environmental Pollutants and Bioavailability  12838   
12846                   Food Science of Animal Resources  12838   
12847                                       Geochemistry  12838   
12848                      Journal of Computer Languages  12838   
12849                    Mining Metallurgy & Exploration  12838   
12850  Photobiomodulation Photomedicine and Laser Sur...  12838   
12851          Polymer-Plastics Technology and Materials  12838   
12852             Sexual and Reproductive Health Matters  12838   
12853                      Woun

## Drop rows with invalid impact factors
18 journals have no impact factor available. Since impact factor is the main property we will look at, we removed these rows

In [106]:
to_drop = list(journal_clean[journal_clean['Impact_Factor'] == 'Not Available'].index.values)
journal_clean = journal_clean.drop(to_drop).reset_index(drop = True)

## Convert types and make title lowercase
- Most columns with floats as values have the type object. We replaced 'Not Available' values with NaN and converted these columns to floats
- Make the title column lowercase for consistency

In [107]:
journal_clean.dtypes

Title                 object
Rank                   int64
Impact_Factor         object
5_Yr_IF               object
Half_Life             object
Percentage_Citable    object
Avg_IF_Percentile     object
dtype: object

In [108]:
# convert the impact factor column
journal_clean['Impact_Factor'] = journal_clean['Impact_Factor'].astype(float)

In [109]:
# convert other columns (that have 'Not Available' values) to float
def f(x):
    try:
        return np.float(x)
    except:
        return np.nan

convert_attr = ['5_Yr_IF', 'Half_Life', 'Percentage_Citable', 'Avg_IF_Percentile']
for i in convert_attr:
    journal_clean[i] = journal_clean[i].apply(f)

In [112]:
# convert title column to lowercase
journal_clean.loc[:, 'Title'] = journal_clean.loc[:, 'Title'].str.lower()

## Export as csv file

In [113]:
journal_clean

Title   Rank  \
0                     ca-a cancer journal for clinicians      1   
1                        new england journal of medicine      2   
2                               nature reviews materials      3   
3                          nature reviews drug discovery      4   
4                                                 lancet      5   
5                            who technical report series      6   
6                  nature reviews molecular cell biology      7   
7                       nature reviews clinical oncology      8   
8                                  nature reviews cancer      9   
9                                       chemical reviews     10   
10                                         nature energy     11   
11      jama-journal of the american medical association     12   
12                             reviews of modern physics     13   
13                              chemical society reviews     14   
14                                                nature     15   
15                                               science     16   
16                        nature reviews disease primers     17   
17                                      world psychiatry     18   
18                                      world psychiatry     18   
19                             nature reviews immunology     20   
20                                      nature materials     21   
21                                                  cell     22   
22                                  nature biotechnology     23   
23                                       nature medicine     24   
24                          living reviews in relativity     25   
25                              nature reviews chemistry     26   
26                           nature reviews microbiology     27   
27                                       lancet oncology     28   
28                           nature reviews neuroscience     29   
29                               nature reviews genetics     30   
...                                                  ...    ...   
12807     hong kong journal of dermatology & venereology  12808   
12808                                 language & history  12808   
12809                              international surgery  12810   
12810                    studies in symbolic interaction  12810   
12811                                  oil & gas journal  12812   
12812                university of pittsburgh law review  12813   
12813                              britain and the world  12814   
12814                                          oncologie  12815   
12815                  securities regulation law journal  12816   
12816                                     zeitgeschichte  12817   
12817                    deutsche lebensmittel-rundschau  12818   
12818                            psychiatrie de l enfant  12819   
12819                            psychiatrie de l enfant  12819   
12820                         zeitschrift fur ethnologie  12821   
12821                                  aerospace america  12822   
12822                          manufacturing engineering  12823   
12823                               amme idaresi dergisi  12824   
12824                  wochenblatt fur papierfabrikation  12824   
12825  correspondances en metabolismes hormones diabe...  12826   
12826                                    natural history  12827   
12827                               acta physica slovaca  12828   
12828                        bell labs technical journal  12828   
12829     bollettino di storia delle scienze matematiche  12828   
12830              current perspectives in social theory  12828   
12831           journal of neurological sciences-turkish  12828   
12832                                pulp & paper-canada  12828   
12833                                    sen-i gakkaishi  12828   
12834                     surface coatings international  12828   
12835                          

In [114]:
journal_clean.to_csv('journal_dataset.csv', index = False)